internal/controller/telemetry/collector_controller.go

```diff
@@ -563,16 +563,72 @@ func normalizeStringSlice(inputList []string) []string {
 	return uniqueList
 }
 
-func setOtelCommonFieldsDefaults(otelCommonFields *otelv1beta1.OpenTelemetryCommonFields, additionalArgs map[string]string, saName string) error {
+func appendAdditionalVolumesForTenantsFileStorage(otelCommonFields *otelv1beta1.OpenTelemetryCommonFields, tenants []v1alpha1.Tenant) {
+	var volumeMountsForInit []corev1.VolumeMount
+	var chmodCommands []string
+
+	for _, tenant := range tenants {
+		if tenant.Spec.PersistenceConfig.EnableFileStorage {
+			bufferVolumeName := fmt.Sprintf("buffervolume-%s", tenant.Name)
+			mountPath := storage.DetermineFileStorageDirectory(tenant.Spec.PersistenceConfig.Directory)
```

Comment:
we need this to be isolated per tenant by default

Reply:
Added, manually tested: 
![kép](https://github.com/user-attachments/assets/c51a8cc5-5b34-49d8-acd2-c3af5c0f5a8c)